cmd/restic-wrapper/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	corev1api "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+func main() {
+	repo := repoName(os.Args[1:])
+	if repo == "" {
+		checkError(errors.New("repository flag (-r) not found"), "ERROR getting repository name")
+	}
+
+	ns := os.Getenv("HEPTIO_ARK_NAMESPACE")
+	if ns == "" {
+		checkError(errors.New("HEPTIO_ARK_NAMESPACE environment variable not found"), "ERROR getting heptio ark namespace")
+	}
+
+	secrets, err := secretInterface(ns)
+	checkError(err, "ERROR getting secrets interface")
+
+	secret, err := secrets.Get("restic-credentials", metav1.GetOptions{})
+	checkError(err, "ERROR getting heptio-ark/restic-credentials secret")
+
+	passwordFile, err := createPasswordFile(secret, repo)
+	checkError(err, "ERROR creating password temp file")
+	defer os.Remove(passwordFile)
+
+	// set the env vars that restic uses for azure credentials
+	os.Setenv("AZURE_ACCOUNT_NAME", os.Getenv("AZURE_STORAGE_ACCOUNT_ID"))
+	os.Setenv("AZURE_ACCOUNT_KEY", os.Getenv("AZURE_STORAGE_KEY"))
+
+	resticArgs := append([]string{"-p=" + passwordFile}, os.Args[1:]...)
```

Comment:
If there are longhand forms for the flags, it might be clearer to use those.

Reply:
good idea.